Source/items.cpp

```diff
@@ -33,6 +33,447 @@ namespace devilution {
 namespace {
 std::optional<CelSprite> itemanims[ITEMTYPES];
 
+int RndPL(int param1, int param2)
+{
+    return param1 + GenerateRnd(param2 - param1 + 1);
+}
+
+int PLVal(int pv, int p1, int p2, int minv, int maxv)
+{
+    if (p1 == p2)
+        return minv;
+    if (minv == maxv)
+        return minv;
+    return minv + (maxv - minv) * (100 * (pv - p1) / (p2 - p1)) / 100;
+}
+
+struct SaveItemPowerParams
+{
+    item_effect_type power;
+    int randMin;
+    int randMax;
+    int minval;
+    int maxval;
+    int multval;
+};
+
+SaveItemPowerParams GetPrefixPowerParams(const PLStruct& prefixParams)

```

Comment:
Maybe it would make sens to make this a member of `SaveItemPowerParams`:

`auto saveParams = SaveItemPowerParams::fromPowerList(PL_Prefix[preidx]);`

Reply:
I've moded both the "Get***Params(...)" to static builder methods. Personally I like to keep structs very bare and have free functions instead (unless we truly need a class), but in this case the static methods don't hurt readability IMO.

Recently I've become one of those devs who think three times before making member functions to structs or even creating classes. People seem to overuse them quite a bit, and nowadays the OOP norm makes people encapsulate everything into classes regardless of whether they truly have shared state or not. These things (polymorphic classes, overused OOP) do have costs in the compiled code, so it's always good to be aware of that. 

Fortunately, the point above doesn't apply here as we're only adding them as builder statics, just wanted to rant at OOP for a bit 😆

Reply:
Diablo has to little OOP :)